sv.c

```diff
@@ -2418,7 +2419,7 @@ S_sv_2iuv_common(pTHX_ SV *const sv)
             if (ckWARN(WARN_NUMERIC) && ((numtype & IS_NUMBER_TRAILING)))
                 not_a_number(sv);
             S_sv_setnv(aTHX_ sv, numtype);
-            return FALSE;
+            goto got_nv;        /* Fill IV/UV slot and set IOKp */
```

Comment:
I appreciate this should be safe with current core, but it takes some thinking to validate that and feels like a trap that could cause problems in the future. A viable alternative might be to duplicate the `(void)SvIOKp_on(sv)` and move the label to line 2353 (ie the `else` branch of `if (SvNVX(sv) < (NV)IV_MAX + 0.5)`.

Reply:
Ah you're quite right, I failed to pick up the right bits of context; looking at the function as a whole this makes sense.